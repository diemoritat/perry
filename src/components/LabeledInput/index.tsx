import { h } from "preact"
import Input from "rebass/dist/Input"
import Label from "rebass/dist/Label"

interface InputProps {
  label?: string
  [a: string]: any
}

export const LabeledInput = (props: InputProps) => (
  <div>
    <Label>{props.label}</Label>
    <Input {...props} />
  </div>
)

export default LabeledInput
