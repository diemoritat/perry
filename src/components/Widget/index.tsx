import { h } from "preact"

export interface WidgetProps {
  color?: string;
};

const Widget = (props: WidgetProps) => (
  <h1 style={{ color: props.color }}>
    Bug Reporter Widget
  </h1>
);

export default Widget;