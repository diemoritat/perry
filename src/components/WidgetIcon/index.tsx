import { h } from "preact"
import BugIcon from "../BugIcon"
import { WidgetStatus } from "../Widget"
import Circle from "rebass/dist/Circle"
import Base from "rebass/dist/Base"

interface WidgetIconProps {
  status: WidgetStatus
}

const RecordingIcon = Circle.extend`
  background-color: #f00;
  margin: 0;
`
const StoppedIcon = Base.extend`
  width: 24px;
  height: 24px;
  display: block;
  background-color: #f00;
`

const IconContainer = Base.extend`
  margin-right: 5px;
`

const WidgetIcon = (props: WidgetIconProps) => {
  switch (props.status) {
    case WidgetStatus.IDLE:
      return (
        <IconContainer>
          <BugIcon />
        </IconContainer>
      )
    case WidgetStatus.RECORDING:
      return (
        <IconContainer>
          <RecordingIcon size={24} />
        </IconContainer>
      )
    case WidgetStatus.STOPPED:
      return (
        <IconContainer>
          <StoppedIcon />
        </IconContainer>
      )
  }
}

export default WidgetIcon
