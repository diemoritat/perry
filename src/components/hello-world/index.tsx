import { h, Component } from "preact"

export default class App extends Component {
  render(props) {
    return (
      <div>
        <h1 style={{ color: props.color }}>Hello, World!</h1>
      </div>
    )
  }
}
