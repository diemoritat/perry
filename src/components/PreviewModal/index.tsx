import { h } from "preact"

import Modal from "rebass/dist/Modal"
import Heading from "rebass/dist/Heading"
import Box from "grid-styled/dist/Box"
import Base from "rebass/dist/Base"
import Close from "rebass/dist/Close"
import LabeledInput from "../LabeledInput"
import Button from "rebass/dist/Button"
import PerryReport from "@/interfaces/PerryReport"

interface ReportForm {
  title: string
  description: string
  screenshotUrl?: string
}

interface ModalProps {
  isOpen: boolean
  onClose: () => void
  onSubmit: () => void
  onDiscard: () => void
  form: ReportForm
  updateForm: (prop: string) => void
}

const Overlay = Base.extend`
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  z-index: -1;
`

const StyledModal = Modal.extend`
  padding: 40px;
  width: 60%;
`

const PreviewModal = (props: ModalProps) =>
  props.isOpen && (
    <div>
      <StyledModal width={256}>
        <Box>
          <Close onClick={props.onClose} />
          <Heading fontSize={14}>Submit report</Heading>
          <LabeledInput
            label="Report title"
            value={props.form.title}
            onInput={props.updateForm("title")}
          />
          <LabeledInput
            label="Description or comments"
            value={props.form.title}
            onInput={props.updateForm("description")}
          />
        </Box>
        <Button onClick={props.onSubmit}>Submit</Button>
        <Button onClick={props.onDiscard}>Discard</Button>
      </StyledModal>
      <Overlay onClick={props.onClose} />
    </div>
  )

export default PreviewModal
