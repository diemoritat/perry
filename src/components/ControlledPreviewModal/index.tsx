import { h, Component } from "preact"
import * as PropTypes from "prop-types";
import PreviewModal from "@/components/PreviewModal";
import PerryReportInfo from "@/interfaces/PerryReportInfo";

interface ControlledPreviewModalProps {
  open: boolean;
  onSubmit?: (info: PerryReportInfo) => void;
  onDiscard?: () => void;
}

export default class ControlledPreviewModal extends Component<ControlledPreviewModalProps, PerryReportInfo> {
  state = {
    title: "",
    description: "",
    screenshotUrl: ""
  };

  static defaultProps = {
    open: false,
    onSubmit: () => {},
    onDiscard: () => {}
  };

  static propTypes = {
    open: PropTypes.bool,
    onSubmit: PropTypes.func,
    onDiscard: PropTypes.func,
  };

  handleFieldChange = ({ target: { name, value }}: any) =>
    this.setState({ [name]: value });

  render() {
    const {
      open,
      onSubmit,
      onDiscard,
    } = this.props;

    return open && (
      <PreviewModal
        form={this.state}
        onSubmit={() => onSubmit(this.state)}
        onDiscard={onDiscard}
        onFieldChange={this.handleFieldChange}
      />
    );
  }
}