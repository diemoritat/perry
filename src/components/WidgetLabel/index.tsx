import { h } from "preact"
import Text from "rebass/dist/Text"
import { WidgetState, WidgetButton } from "@/components/Widget"

interface WidgetLabelProps {
  state: WidgetState
}

const getLabelForState = (state: WidgetState) => {
  switch (state) {
    case WidgetState.IDLE:
      return "Start recording"

    case WidgetState.RECORDING:
      return "Recording..."

    case WidgetState.STOPPED:
      return "Submit"
  }
}

const Label = Text.extend`
  color: #444;
  opacity: 0;
  transition: all 0.4s;
  white-space: nowrap;
  margin-bottom: 3px;
  margin-right: 3px;

  ${WidgetButton}:hover & {
    opacity: 1;
  }
`

const WidgetLabel = (props: WidgetLabelProps) => <Label>{getLabelForState(props.state)}</Label>

export default WidgetLabel
