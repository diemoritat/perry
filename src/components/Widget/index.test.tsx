/* eslint-env node, jest */

import { h } from "preact"
import { render } from "preact-render-to-string"
import { WidgetState } from "@/components/Widget"

import defaultOptions from "../../packages/default-options"

import Widget from "./"

describe("Widget Snapshot", () => {
  it("should render with content", () => {
    const tree = render(<Widget {...defaultOptions} />)
    expect(tree).toMatchSnapshot()
  })
})
