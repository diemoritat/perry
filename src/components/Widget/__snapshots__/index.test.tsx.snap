// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget Snapshot should render with content 1`] = `"<h1>Bug Reporter Widget</h1>"`;
